qiime/core/testing/format.py

```diff
@@ -0,0 +1,61 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from qiime.plugin import TextFileFormat
+import qiime.plugin.resource as resource
+
+
+class IntSequenceFormat(TextFileFormat):
+    def sniff(self):
+        with self.open() as fh:
+            for line, _ in zip(fh, range(5)):
+                try:
+                    int(line.rstrip('\n'))
+                except (TypeError, ValueError):
+                    return False
+            return True
+
+
+class MappingFormat(TextFileFormat):
+    def sniff(self):
+        with self.open() as fh:
+            for line, _ in zip(fh, range(5)):
+                cells = line.rstrip('\n').split('\t')
+                if len(cells) != 2:
+                    return False
+            return True
+
+
+class SingleIntFormat(TextFileFormat):
+    def sniff(self):
+        with self.open() as fh:
+            try:
+                int(fh.readline().rstrip('\n'))
+            except (TypeError, ValueError):
+                return False
+        return True
+
+
+IntSequenceDirectoryFormat = resource.SingleFileDirectoryFormat(
+    'IntSequenceDirectoryFormat', 'ints.txt', IntSequenceFormat)
+
+
+class MappingDirectoryFormat(resource.DirectoryFormat):
```

Comment:
Add comment explaining that this could have been a `SingleFileDirectoryFormat`, but isn't for testing purposes.